src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -3472,6 +3504,43 @@ internal static bool IsTypeParameterizedProperty(Type t)
             return null;
         }
 
+        private protected T GetDotNetProperty<T>(object obj, MemberNamePredicate predicate) where T : PSMemberInfo
+        {
+            bool lookingForProperties = typeof(T).IsAssignableFrom(typeof(PSProperty));
+            bool lookingForParameterizedProperties = IsTypeParameterizedProperty(typeof(T));
+            if (!lookingForProperties && !lookingForParameterizedProperties)
+            {
+                return null;
+            }
+
+            CacheTable typeTable = _isStatic
+                ? GetStaticPropertyReflectionTable((Type)obj)
+                : GetInstancePropertyReflectionTable(obj.GetType());
+
+            object entry = typeTable.GetOrDefault(predicate);
+            if (entry == null)
+            {
+                return null;
+            }
+
+            if (entry is PropertyCacheEntry propertyEntry && lookingForProperties)
+            {
+                var isHidden = propertyEntry.member.GetCustomAttributes(typeof(HiddenAttribute), false).Any();
+                return new PSProperty(propertyEntry.member.Name, this, obj, propertyEntry) { IsHidden = isHidden } as T;
+            }
+
+            if (entry is ParameterizedPropertyCacheEntry parameterizedPropertyEntry && lookingForParameterizedProperties)
+            {
+                // TODO: check for HiddenAttribute
+                // We can't currently write a parameterized property in a PowerShell class so this isn't too important,
+                // but if someone added the attribute to their C#, it'd be good to set isHidden correctly here.
+                return new PSParameterizedProperty(parameterizedPropertyEntry.propertyName,
```

Comment:
The common pattern for method declarations spread across multiple lines is to place arguments starting on the next line.
```suggestion
                return new PSParameterizedProperty(
                    parameterizedPropertyEntry.propertyName,
```